.github/workflows/checkstyle.yml

```diff
@@ -0,0 +1,52 @@
+name: Run Checkstyle
+
+on:
+  pull_request:
+    branches:
+      - "**"
+
+jobs:
+  checkstyle:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: temurin
+
+      - name: Cache Maven dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-m2
+
+      # Install dependencies and handle compilation errors
+      - name: Install dependencies
+        run: |
+          # First check if the code compiles
+          mvn compile -DskipTests
+          if [ $? -eq 0 ]; then
+            # If compilation is successful, install dependencies
+            mvn install -DskipTests
+          else
+            # If compilation fails, output error message and stop the workflow
+            echo "Compilation failed! Fix compilation errors before running Checkstyle."
+            exit 1
+          fi
+
+      # Run Checkstyle on changed files
+      - name: Run Checkstyle on changed files
+        run: |
+          CHANGED_FILES=$(git diff --name-only ${{ github.event.before }} ${{ github.sha }} | grep '\.java$')
```

Comment:
Are you sure github.sha will work?
This hash is not a commit that exists in the repo. 
We had issues with eslint action.

https://github.com/PSMRI/.github/blob/main/.github/workflows/lint_audit.yml#L43

Reply:
for this i'll avoid using git diff and configure checkstyle workflow to scan entire codebase